File: Trace/UI/Theme.swift
import SwiftUI

extension Color {
    static let traceParchment = Color(red: 0.96, green: 0.94, blue: 0.89)
    static let traceCream = Color(red: 0.985, green: 0.965, blue: 0.93)
    static let traceInk = Color(red: 0.13, green: 0.11, blue: 0.10)
    static let traceMuted = Color(red: 0.42, green: 0.37, blue: 0.32)
    static let traceAccent = Color(red: 0.61, green: 0.26, blue: 0.13)
    static let traceHair = Color(red: 0.78, green: 0.73, blue: 0.65)
}

extension Font {
    static let traceDisplay = Font.system(size: 34, weight: .semibold, design: .serif)
    static let traceTitle = Font.system(size: 26, weight: .semibold, design: .serif)
    static let traceCardTitle = Font.system(size: 22, weight: .semibold, design: .serif)
    static let traceBody = Font.system(size: 17, weight: .regular, design: .serif)
    static let traceSubtitle = Font.system(size: 15, weight: .regular, design: .default)
    static let traceLabel = Font.system(size: 12, weight: .medium, design: .default)
}

struct VibeGradient {
    static func colors(for vibes: [Vibe]) -> [Color] {
        let primary = vibes.first ?? .urban
        switch primary {
        case .historic: return [Color(red: 0.55, green: 0.35, blue: 0.20), Color(red: 0.30, green: 0.18, blue: 0.10)]
        case .nature:   return [Color(red: 0.32, green: 0.50, blue: 0.32), Color(red: 0.12, green: 0.25, blue: 0.15)]
        case .weird:    return [Color(red: 0.45, green: 0.30, blue: 0.55), Color(red: 0.22, green: 0.12, blue: 0.30)]
        case .coffee:   return [Color(red: 0.50, green: 0.32, blue: 0.22), Color(red: 0.28, green: 0.18, blue: 0.12)]
        case .iconic:   return [Color(red: 0.70, green: 0.40, blue: 0.20), Color(red: 0.40, green: 0.20, blue: 0.10)]
        case .hidden:   return [Color(red: 0.30, green: 0.28, blue: 0.32), Color(red: 0.12, green: 0.12, blue: 0.15)]
        case .water:    return [Color(red: 0.25, green: 0.40, blue: 0.55), Color(red: 0.10, green: 0.20, blue: 0.32)]
        case .urban:    return [Color(red: 0.40, green: 0.36, blue: 0.32), Color(red: 0.20, green: 0.18, blue: 0.16)]
        }
    }
}

File: mobile/src/theme/theme.ts
import type { Vibe } from '../models/run';

/**
 * Palette and typography for Trace. Mirrors the SwiftUI Theme: a parchment /
 * cream editorial look with a burnt-sienna accent, serif for body + titles,
 * system sans for labels.
 */
export const colors = {
  parchment: '#F5F0E3', // 0.96, 0.94, 0.89
  cream: '#FBF6ED', //     0.985, 0.965, 0.93
  ink: '#211C19', //       0.13, 0.11, 0.10
  muted: '#6B5E52', //     0.42, 0.37, 0.32
  accent: '#9C4221', //    0.61, 0.26, 0.13
  hair: '#C7BAA6', //      0.78, 0.73, 0.65
  white: '#FFFFFF',
  black: '#000000',
} as const;

/**
 * Font families loaded via @expo-google-fonts/lora in App.tsx.
 * Lora is a well-balanced contemporary serif — the cross-platform stand-in for
 * the SwiftUI "New York" serif used in the native build.
 */
export const fonts = {
  serifRegular: 'Lora_400Regular',
  serifMedium: 'Lora_500Medium',
  serifSemiBold: 'Lora_600SemiBold',
  // Labels/eyebrows use the platform sans (San Francisco / Roboto) for contrast.
  sans: undefined as string | undefined,
} as const;

/** Type scale mirroring the SwiftUI Font extension. */
export const type = {
  display: { fontFamily: fonts.serifSemiBold, fontSize: 32 },
  title: { fontFamily: fonts.serifSemiBold, fontSize: 26 },
  cardTitle: { fontFamily: fonts.serifSemiBold, fontSize: 22 },
  body: { fontFamily: fonts.serifRegular, fontSize: 17 },
  subtitle: { fontFamily: fonts.sans, fontSize: 15 },
  label: { fontFamily: fonts.sans, fontSize: 12, fontWeight: '600' as const },
} as const;

/** Two-stop gradient keyed to a run's primary vibe (hero placeholder). */
export function vibeGradient(vibes: Vibe[]): [string, string] {
  const primary = vibes[0] ?? 'urban';
  switch (primary) {
    case 'historic':
      return ['#8C5933', '#4D2E1A'];
    case 'nature':
      return ['#528052', '#1F4026'];
    case 'weird':
      return ['#734D8C', '#381F4D'];
    case 'coffee':
      return ['#805238', '#472E1F'];
    case 'iconic':
      return ['#B36633', '#66331A'];
    case 'hidden':
      return ['#4D4752', '#1F1F26'];
    case 'water':
      return ['#40668C', '#1A3352'];
    case 'urban':
      return ['#665C52', '#332E29'];
  }
}
